espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/InvocationOperation.java

```diff
@@ -0,0 +1,376 @@
+
+                if (method == null) {
+                    StringBuilder stringBuilder = new StringBuilder("No such method found: ");
+                    stringBuilder.append(InvocationOperation.this.methodName);
+                    stringBuilder.append("(");
+                    int length = InvocationOperation.this.arguments.size();
+
+                    for (int i = 0; i < length; i++) {
+                        Object argument = InvocationOperation.this.arguments.get(i);
+
+                        if (i != 0) {
+                            stringBuilder.append(", ");
+                        }
+
+                        String simpleName;
+
+                        if (argument == null) {
+                            simpleName = "null";
+                        } else {
+                            simpleName = argument.getClass().getSimpleName();
+                        }
+
+                        stringBuilder.append("[").append(simpleName).append("]");
+                    }
+
+                    stringBuilder.append(")");
+
+                    ref.set(BackdoorResultVoid.instance
+                            .asMap(InvocationOperation.this.methodName, o, stringBuilder.toString()));
+                    return;
+                }
+
+                Object result;
+
+                try {
+                    if (method.getMethod().getReturnType().equals(Void.TYPE)) {
+                        method.invoke(o);
+                        result = "<VOID>";
+                    } else {
+                        result = method.invoke(o);
+                    }
+
+                    ref.set(result);
+                    return;
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    refEx.set(e);
+                    return;
+                }
+            }
+        });
+
+        if (refEx.get() != null) {
+            throw refEx.get();
+        }
+
+        return ref.get();
+    }
+
+    public MethodWithArguments findCompatibleMethod(Object object) {
+        return findCompatibleMethod(object.getClass());
+    }
+
+    public MethodWithArguments findCompatibleMethod(Class<?> forClass) {
+        // Fast path
+        try {
+            return new MethodWithArguments(forClass.getMethod(methodName, argumentTypes),
+                    new ArrayList<Object>(arguments));
+        } catch (NoSuchMethodException e) {
+            // No immediate method found
+
+            for (Method method : forClass.getMethods()) {
+                if (!method.getName().equals(methodName)) {
+                    continue;
+                }
+
+                try {
+                    return new MethodWithArguments(method, parseToSuitableArguments(method));
+                } catch (IncompatibleArgumentsException e1) {
+                    continue;
+                }
+            }
+
+            return null;
+        }
+    }
+
+    private static Class<?> mapToPrimitiveClass(Class<?> c) {
```

Comment:
these helper methods should be moved to a separate utility/helper class. It would be easier to cover them with unit tests this way

Reply:
moved to another utility class